Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp

```diff
@@ -4,23 +4,120 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/Debug.h>
 #include <LibGfx/ImageFormats/JBIG2Loader.h>
 
 // Spec: ITU-T_T_88__08_2018.pdf in the zip file here:
 // https://www.itu.int/rec/T-REC-T.88-201808-I
 
 namespace Gfx {
 
+// JBIG2 spec, Annex D, D.4.1 ID string
+static constexpr u8 id_string[] = { 0x97, 0x4A, 0x42, 0x32, 0x0D, 0x0A, 0x1A, 0x0A };
+
+// Annex D
+enum class Organization {
+    // D.1 Sequential organization
+    Sequential,
+
+    // D.2 Random-access organization
+    RandomAccess,
+
+    // D.3 Embedded organization
+    Embedded,
+};
+
+struct JBIG2LoadingContext {
+    enum State {
+        NotDecoded = 0,
+        Error,
+    };
+    State state { State::NotDecoded };
+    ReadonlyBytes data;
+
+    Organization organization { Organization::Sequential };
+    IntSize size;
+
+    Optional<u32> number_of_pages;
+};
+
+static bool decode_jbig2_header(JBIG2LoadingContext& context)
+{
+    if (!JBIG2ImageDecoderPlugin::sniff(context.data)) {
+        dbgln_if(JBIG2_DEBUG, "Invalid JBIG2 header");
+        context.state = JBIG2LoadingContext::State::Error;
+        return false;
+    }
```

Comment:
```cpp
if (!JBIG2ImageDecoderPlugin::sniff(context.data))
    return Error::from_string_literal("Invalid JBIG2 header");
```
There is no need to set the state, as plugins associated to a file with an invalid header do not exist.